src/redis-cli.c

```diff
@@ -5864,6 +5865,26 @@ static int clusterManagerCommandAddNode(int argc, char **argv) {
     listAddNodeTail(cluster_manager.nodes, new_node);
     added = 1;
 
+    if (!master_node) {
+        // Send functions to the new node, if new node is a replica it will get the functions from his primary.
+        clusterManagerLogInfo(">>> Getting functions from cluster\n");
+        reply = CLUSTER_MANAGER_COMMAND(refnode, "FUNCTION DUMP");
+        if (!clusterManagerCheckRedisReply(refnode, reply, &err)) {
+            clusterManagerLogInfo(">>> Failed retrieving Functions from the cluster, "
+                    "skip this step as Redis version do not support function command (error = '%s')\n", err? err : "NULL reply");
+            if (err) zfree(err);
+        } else {
+            assert(reply->type == REDIS_REPLY_STRING);
+            clusterManagerLogInfo(">>> Send FUNCTION RESTORE to %s:%d\n", ip, port);
+            function_restore_reply = CLUSTER_MANAGER_COMMAND(new_node, "FUNCTION RESTORE %b", reply->str, reply->len);
```

Comment:
We should probably check that there are no existing functions, like we do with data in clusterManagerNodeIsEmpty, or multi-exec the a flush along with this.

Reply:
since this dump contains multiple functions (or soon to be libraries), I wanna extend FUNCTION RESTORE with some argument like:
FLUSH - completely replace all existing functions
APPEND - add non-existing ones, and fail if there are conflicts
REPLACE - add non-existing ones, and override existing ones.

in this case, add-node can use FLUSH

@madolson @yossigo WDYT?

Reply:
@oranagra I think it makes sense.

Reply:
@oranagra fixed accordingly and updated top comment.